src/Stream/src/StreamBase/Extensions/StreamServicesExtensions.cs

```diff
@@ -105,5 +110,36 @@ public static void AddStreamServices(this IServiceCollection services, IConfigur
 
             services.AddStreamCoreServices(configuration);
         }
+
+        public static void AddStreamServices<T>(this IServiceCollection services, IConfiguration configuration)
+        {
+            services.AddOptions();
+            services.SafeAddRabbitMQConnection(configuration);
+            services.AddRabbitConnectionFactory();
+            services.ConfigureRabbitOptions(configuration);
+            services.AddSingleton<IApplicationContext, GenericApplicationContext>();
+
+            services.AddStreamServices(configuration);
+            services.AddSourceStreamBinding();
+            services.AddSinkStreamBinding();
+
+            services.AddEnableBinding<T>();
+        }
+
+        // Deal with intermittent TypeLoadExceptions in Connectors (mac, linux)s
+        public static void SafeAddRabbitMQConnection(this IServiceCollection services, IConfiguration configuration)
+        {
+            var loggerFactory = services.BuildServiceProvider().GetService<ILoggerFactory>();
+            var logger = loggerFactory?.CreateLogger("Steeltoe.Stream.Extensions.SafeAddRabbitMQConnection");
+
+            try
+            {
+                services.AddRabbitMQConnection(configuration);
+            }
+            catch (Exception ex)
+            {
+                logger?.LogDebug(ex, ex.Message);
```

Comment:
Should this be logging as a warning instead of just debug?

Reply:
Sure yes, I had it originally as error, but wasn't sure if it was somehow tripping the test runner.